### ISSUE TITLE
getting "ValueError: No arrays to thread error on to." 

### ISSUE THREAD

**Issue by Negar-Erfanian** [patrick-kidger/equinox#438]
Hi Patrick,

I wrote an ode solver code using diffrax 2 months back and it was working. 
However, after two months that I am back to the code it gives me this error : "ValueError: No arrays to thread error on to." which raises when using diffrax.diffeqsolve.

I wrote this based on your own example in [here](https://docs.kidger.site/diffrax/examples/neural_ode/).
Therefore I went back to your own example and ran it and again it gave me the same error:

```
ValueError                                Traceback (most recent call last)
[<ipython-input-4-57a4e3ced7d9>](https://localhost:8080/#) in <cell line: 140>()
    138 
    139     return ts, ys, model
--> 140 ts, ys, model = main()

5 frames
[<ipython-input-4-57a4e3ced7d9>](https://localhost:8080/#) in main(dataset_size, batch_size, lr_strategy, steps_strategy, length_strategy, width_size, depth, seed, plot, print_every)
     89     data_key, model_key, loader_key = jrandom.split(key, 3)
     90 
---> 91     ts, ys = get_data(dataset_size, key=data_key)
     92     _, length_size, data_size = ys.shape
     93 

[<ipython-input-4-57a4e3ced7d9>](https://localhost:8080/#) in get_data(dataset_size, key)
     56     ts = jnp.linspace(0, 10, 100)
     57     key = jrandom.split(key, dataset_size)
---> 58     ys = jax.vmap(lambda key: _get_data(ts, key=key))(key)
     59     return ts, ys
     60 

    [... skipping hidden 3 frame]

[<ipython-input-4-57a4e3ced7d9>](https://localhost:8080/#) in <lambda>(key)
     56     ts = jnp.linspace(0, 10, 100)
     57     key = jrandom.split(key, dataset_size)
---> 58     ys = jax.vmap(lambda key: _get_data(ts, key=key))(key)
     59     return ts, ys
     60 

[<ipython-input-4-57a4e3ced7d9>](https://localhost:8080/#) in _get_data(ts, key)
     46     dt0 = 0.1
     47     saveat = diffrax.SaveAt(ts=ts)
---> 48     sol = diffrax.diffeqsolve(
     49         diffrax.ODETerm(f), solver, ts[0], ts[-1], dt0, y0, saveat=saveat
     50     )

    [... skipping hidden 15 frame]

[/usr/local/lib/python3.10/dist-packages/diffrax/integrate.py](https://localhost:8080/#) in diffeqsolve(terms, solver, t0, t1, dt0, y0, args, saveat, stepsize_controller, adjoint, discrete_terminating_event, max_steps, throw, solver_state, controller_state, made_jump)
    570         with jax.ensure_compile_time_eval():
    571             pred = (t1 - t0) * dt0 < 0
--> 572         dt0 = eqxi.error_if(dt0, pred, msg)
    573 
    574     # Backward compatibility

    [... skipping hidden 1 frame]

[/usr/local/lib/python3.10/dist-packages/equinox/_errors.py](https://localhost:8080/#) in branched_error_if(x, pred, index, msgs, on_error)
    262     flat = jtu.tree_leaves(dynamic_x)
    263     if len(flat) == 0:
--> 264         raise ValueError("No arrays to thread error on to.")
    265     dynamic_x = _error(dynamic_x, pred, index, msgs=msgs, on_error=on_error)
    266     return combine(dynamic_x, static_x)

ValueError: No arrays to thread error on to.
```

Do you know why this is happening?
I also got this warning when installing equinox:

```
/usr/local/lib/python3.10/dist-packages/equinox/_ad.py:753: UserWarning: As of Equinox 0.10.7, `equinox.filter_custom_vjp.defvjp` is deprecated in favour of `.def_fwd` and `.def_bwd`. This new API supports symbolic zeros, which allow for more efficient autodifferentiation rules. In particular:
- the fwd and bwd functions take an extra `perturbed` argument, which     indicates which primals actually need a gradient. You can use this     to skip computing the gradient for any unperturbed value. (You can     also safely just ignore this if you wish.)
- `None` was previously passed to indicate a symbolic zero gradient for     all objects that weren't inexact arrays, but all inexact arrays     always had an array-valued gradient. Now, `None` may also be passed     to indicate that an inexact array has a symbolic zero gradient.
```

**Comment by patrick-kidger** [comment]
Yup! This will be fixed in the next release of Diffrax -- happening in the next few days.
(This was a sneaky bug in Diffrax all along :D )

**Comment by Negar-Erfanian** [comment]
oh ok! Thanks!

Is there a way I could still make it work before the new version releases? like working with previous versions of the current packages. if so please let me know which ones and what versions?

Thank you so much!

**Comment by patrick-kidger** [comment]
Yup! Either downgrade Equinox (I think v0.10.10 should work), or upgrade Diffrax to the current GitHub HEAD.

**Comment by patrick-kidger** [comment]
I just released Diffrax v0.4.1. That should fix this!
